sdks/python/apache_beam/io/gcp/bigtable_io_write.py

```diff
@@ -0,0 +1,166 @@
+    self.written.inc()
+    self.batcher.mutate(row)
+
+  def finish_bundle(self):
+    return self.batcher.flush()
+
+  def display_data(self):
+    return {'projectId': DisplayDataItem(self.beam_options.project_id,
+                                         label='Bigtable Project Id'),
+            'instanceId': DisplayDataItem(self.beam_options.instance_id,
+                                          label='Bigtable Instance Id'),
+            'tableId': DisplayDataItem(self.beam_options.table_id,
+                                       label='Bigtable Table Id'),
+            'bigtableOptions': DisplayDataItem(str(self.beam_options),
+                                               label='Bigtable Options',
+                                               key='bigtableOptions'),
+           }
+
+
+class BigtableConfiguration(object):
+  """ Bigtable configuration variables.
+
+  :type project_id: :class:`str` or :func:`unicode <unicode>`
+  :param project_id: (Optional) The ID of the project which owns the
+    instances, tables and data. If not provided, will
+    attempt to determine from the environment.
+
+  :type instance_id: str
+  :param instance_id: The ID of the instance.
+
+  :type table_id: str
+  :param table_id: The ID of the table.
+  """
+
+  def __init__(self, project_id, instance_id, table_id):
+    self.project_id = project_id
+    self.instance_id = instance_id
+    self.table_id = table_id
+    self.credentials = None
+
+
+class BigtableWriteConfiguration(BigtableConfiguration):
+  """ BigTable Write Configuration Variables.
+
+  :type flush_count: int
+  :param flush_count: (Optional) Max number of rows to flush. If it
+    reaches the max number of rows it calls finish_batch() to mutate the
+    current row batch. Default is FLUSH_COUNT (1000 rows).
+
+  :type max_mutations: int
+  :param max_mutations: (Optional)  Max number of row mutations to flush.
+    If it reaches the max number of row mutations it calls finish_batch()
+    to mutate the current row batch. Default is MAX_MUTATIONS
+    (100000 mutations).
+
+  :type max_row_bytes: int
+  :param max_row_bytes: (Optional) Max number of row mutations size
+    to flush. If it reaches the max number of row mutations size it
+    calls finish_batch() to mutate the current row batch. Default is
+    MAX_ROW_BYTES (5 MB).
+
+  :type app_profile_id: str
+  :param app_profile_id: (Optional) The unique name of the AppProfile.
+  """
+
+  def __init__(self, project_id, instance_id, table_id,
+               flush_count=None, max_row_bytes=None, app_profile_id=None):
+    super(BigtableWriteConfiguration, self).__init__(project_id,
+                                                     instance_id,
+                                                     table_id)
+    self.flush_count = flush_count
+    self.max_row_bytes = max_row_bytes
+    self.app_profile_id = app_profile_id
+
+  def __str__(self):
+    import json
+    ret = {'project_id': self.project_id,
```

Comment:
Why are the other settings on in the string?

Reply:
To set the configuration in the `DisplayDataItem`. Because, if we did not set that, in Dataflow, only shows the name of the Class.